### DISCUSSION TITLE
Look-ahead feature that jumps to the next view slightly before MuseScore finishes playing the current view

### DISCUSSION THREAD

**Discussion by scorster** [musescore/MuseScore#21870] [Ideas]
A look-ahead feature has been requested a number of times on the .org forum. 

[Here's one such post.](https://musescore.org/en/node/330306)

This feature would be enormously helpful for students and practicing musicians ... and for scorists who want read along naturally, just as a player would. 

Most sight-readers can read a number of beats ahead of their play position in a score. But MuseScore rigidly keeps notation locked on screen until every bit of it has been played ... and unfortunately that prevents musicians from reading ahead.

**Comment by worldwideweary** [comment]
Question regarding the idea: 
... would an option to force the currently playing/currently navigated system to be *always on top*/*first* in view, and hence have the following system showing below it, so long as it fit, achieve the desired result?

**Reply by scorster** [reply]
> *... would an option to force the currently playing/currently navigated system to be always on top/first in view*

@ worldWideWeary Thanks for your interest in this!

Responding to your suggestion: If MuseScore always dynamically positions the currently playing system to be topmost in the view, that would indeed allow reading ahead *BUT* at the expense of updating the view each time playback reaches the next system—and that the additional, unnecessary screen updates would be undesirable jarring. That is, if I've understood you correctly ...
 
I think it would be ideal if MuseScore always kept in view **a)** the currently playing system *AND* **b)** the system directly below it. And this may be what you suggested. However, jumps and repeats would still have to occur *x amount* before the notation in the current view completes playing.

**Comment by yoichi02** [comment]
I produce a score animation video, by recording score "played" by MuseScore. To make the video, the window size is adjusted to show only one system (or two) in it. This is because the video is used for practice to sing with the score. In such a video, "look ahead" feature is essential. Currently, when the cursor is get close to the end of the page, the page is abruptly and completely "swapped" with the next "page". This does not help for singing with "look ahead", since I can't see the new page until the swap is completed. If it is a dedicated video creation app, "look ahead" may be implemented by gradually revealing the beginning part of the next page, e.g., by peeling-off, sliding-off, scrolling-off, etc. MuseScore, however, is not a video creation app. So, the easiest way may be wiping-off the old page from right to the left gradually or peeling off from the top right corner, instead of the current page-swapping? Mind you that this is different from what seems to be discussed above, which is, to show the current system always at the "top" of the "page by renewing the scrolling more frequently. In my proposal, "look ahead" (revealing the next page) starts happening when the cursor is close to the end of the stave on the  page (maybe still two measures left?) and the change completes when the cursor hits the last bar. Although I change the page size to suit my singing video, this would work with normal piano score, etc., without changing the page size.

**Comment by MarcSabatella** [comment]
I realize this doesn't cover all cases, but keep in mind you can use continuous view with the smooth panning option to have a view where you can always read ahead - that's often a better way to do things like score videos anyhow.  But if you literally do want to show the full page, then you'd want something else indeed.
